### PR TITLE
Fix tolerance arguments in decomposition_test

### PR DESCRIPTION
**Pull request by mrwojtek**

### CODE REVIEW

**Comment by mrwojtek** [comment]
This fixes #1466 (verified)

File: cirq/optimizers/decompositions_test.py [hunk 1]
```diff
@@ -215,16 +215,18 @@ def test_single_qubit_matrix_to_native_gates_cases(intended_effect):
...
+    atol = 1e-6
+
     intended_effect = cirq.dot(
         cirq.unitary(cirq.Z**(2 * pre_turns)),
         cirq.unitary(cirq.X),
         cirq.unitary(cirq.Z**(2 * post_turns)))
 
     gates = cirq.single_qubit_matrix_to_phased_x_z(
-        intended_effect, atol=0.0001)
+        intended_effect, atol=atol)
```
Comment (Strilanc) [inline cirq/optimizers/decompositions_test.py]: This tolerance refers to a per-gate effect, whereas the other tolerance refers to an aggregated effect. Add a safety factor of 10 to the aggregated one.

**Comment by vtomole** [comment]
@mrwojtek I ran the original test 50,000 times with no failures. How were you able to verify this?

File: cirq/optimizers/decompositions_test.py (hunk 1 above)
Comment (mrwojtek) [reply cirq/optimizers/decompositions_test.py]: Sounds good, done.

**Comment by mrwojtek** [comment]
Victory: I ran this test 100000 times before and after my change. It's not a proof but it failed about 30 times before fixing, and 0 times after fixing.

**Comment by vtomole** [comment]
I managed to run the original 1000000 times with no errors (took 1.5 hours!). I wonder what could be the difference. Could it  different verions of numpy? (I'm on `1.16.2`). Python versions? (i'm on `3.6.7`).

**Comment by mrwojtek** [comment]
Ahh! In that case this is most likely my macOS. I had similar problems with other equality comparisons for some of the Cirq tests (that's why I implemented approx_eq). It works on Linux but not on my macOS.

BTW: It took me about 10min to do the same calculation on my MacBook laptop ;-)

**Pull request merged**